File: lambda/generatePresignUrl.js
const AWS = require('aws-sdk');

exports.handler = async (event) => {
  AWS.config.update({ region: 'us-east-1' });

  const s3 = new AWS.S3();

  const bucketName = 'fajjarnr-report';
  const objectKey = 'report.html';

  const expiration = new Date();
  expiration.setHours(expiration.getHours() + 1);

  const url = s3.getSignedUrl('getObject', {
    Bucket: bucketName,
    Key: objectKey,
    Expires: expiration,
  });

  return url;
};

File: lambda/generate_presign_url.py
import logging
import boto3
from botocore.exceptions import ClientError

s3_client = boto3.client('s3')


def lambda_handler(event, context):
    bucket_name = "fajjarnr-report"
    object_name = "report.html"
    expiration_in_seconds = 120

    try:
        presigned_url_str = s3_client.generate_presigned_url('get_object', Params={
            'Bucket': bucket_name, 'Key': object_name}, ExpiresIn=expiration_in_seconds)
        response = {"presigned_url_str": presigned_url_str}
    except ClientError as e:
        logging.error(e)
        return None

    # The response contains the presigned URL
    return response
